src/openms/include/OpenMS/KERNEL/RangeManager.h

```diff
@@ -28,181 +28,574 @@
 // ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 //
 // --------------------------------------------------------------------------
-// $Maintainer: Timo Sachsenberg$
-// $Authors: Marc Sturm $
+// $Maintainer: Chris Bielow $
+// $Authors: Chris Bielow $
 // --------------------------------------------------------------------------
 
 #pragma once
 
-#include <OpenMS/DATASTRUCTURES/DRange.h>
+#include <OpenMS/config.h>
+#include <OpenMS/CONCEPT/Exception.h>
+
+#include <algorithm> // for min/max
+#include <cassert>
+#include <iosfwd>  // for std::ostream
+
 
 namespace OpenMS
 {
-  /**
-    @brief Handles the management of a position and intensity range.
+  /// Dimensions of data acquisition for MS data
+  enum class MSDim
+  {
+    RT,
+    MZ,
+    INT,
+    IM
+  };
 
-    This is needed for all peak and feature container like Spectrum, MSExperiment and FeatureMap.
-  */
-  template <UInt D>
-  class RangeManager
+  /// Base class for a simple range with minimum and maximum
+  struct OPENMS_DLLAPI RangeBase
   {
-public:
-    /// Dimension of the position range
-    enum {DIMENSION = D};
-    /// Position range type
-    typedef DRange<D> PositionRangeType;
-    /// Position Type
-    typedef DPosition<D> PositionType;
-    /// Intensity range type
-    typedef DRange<1> IntensityRangeType;
+  public:
+    /// Ctor: initialize with empty range
+    RangeBase() = default;
+    /// set min and max; if min > max, then an exception will be thrown
+    RangeBase(const double min, const double max) :
+        min_(min), max_(max)
+    {
+      if (min_ > max_)
+        throw "Invalid initialization of range";
+    }
 
-    /// Default constructor
-    RangeManager() :
-      int_range_(),
-      pos_range_()
-    {}
+    /// make the range empty, i.e. isEmpty() will be true
+    void clear()
+    {
+      *this = RangeBase(); // overwrite with fresh instance
+    }
 
-    /// Copy constructor
-    RangeManager(const RangeManager & rhs) :
-      int_range_(rhs.int_range_),
-      pos_range_(rhs.pos_range_)
-    {}
+    /// is the range empty (i.e. default constructed or cleared using clear())?
+    bool isEmpty() const
+    { // invariant: only possible when default constructed or clear()'ed
+      return min_ > max_; 
```

Comment:
if setMin(x) is called on the default constructed range then isEmpty() = false.
But wouldn't this be an empty range (because min = max)?